R/misc.R

```diff
@@ -859,9 +859,8 @@ check_new_data <- function(req, object, new_data) {
   step_cls <- class(object)[1]
   step_id <- object$id
   cli::cli_abort(
-    "The following required {cli::qty(col_diff)} column{?s} {?is/are} \
-    missing from `new_data` in step '{step_id}': {col_diff}.",
-    class = "new_data_missing_column",
```

Comment:
What was the reason for classing this error previously?

Reply:
this was from before `expect_snapshot()` so they were tested with `expect_error()`. Classed errors were used to make sure the correct error bubbled up. Instead of relying on regex matching.

However it is the only place we use it and only for testing purposes, and for that reason I think we are better off using the unclassed errors with `expect_snapshot()`

also, since we were matching on class, we never checked that the {cli} syntax was correct as we didn't see the error

i'm not opposed to classed errors. And they might even be useful if cases where {tune} could match on them.

Reply:
I'm okay with removing the error class if we are not using it. I see tests in the recipes extension packages that test for that error, but you probably know without checking if they test that a recipe step _from that extension package_ is the one throwing the classed error. 👍

Reply:
I ran a revdepcheck. it is only an issue for package i maintain so i think it is fine :) thanks for pointing it out